crates/vfio-ioctls/src/vfio_device.rs

```diff
@@ -791,7 +791,7 @@ impl VfioDevice {
             .irqs
             .get(&irq_index)
             .ok_or(VfioError::VfioDeviceSetIrq)?;
-        if irq.count < vector {
+        if irq.count <= vector {
             return Err(VfioError::VfioDeviceSetIrq);
```

Comment:
This is a general comment about `VfioError::VfioDeviceSetIrq` being used in too many places, making the error type useless because it's almost impossible to identify where the error came from.
Since you're improving the overall code with this PR, would you mind adding some new error types in order to make the error reporting better?